signalfx-java/src/main/java/com/signalfx/signalflow/ServerSentEventsTransport.java

```diff
@@ -0,0 +1,531 @@
+                    .setConnectionRequestTimeout(this.requestConfig.getConnectionRequestTimeout())
+                    .setConnectTimeout(this.requestConfig.getConnectTimeout())
+                    .setProxy(this.requestConfig.getProxy()).build();
+
+            if (log.isDebugEnabled()) {
+                log.debug("constructed request config: " + this.transportRequestConfig.toString());
+            }
+        }
+
+        public CloseableHttpResponse post(String token, String path,
+                                          final Map<String, String> parameters, String body)
+                throws SignalFlowException {
+
+            HttpPost httpPost = null;
+            try {
+                List<NameValuePair> params = new ArrayList<NameValuePair>();
+                if (parameters != null) {
+                    for (Map.Entry<String, String> entry : parameters.entrySet()) {
+                        params.add(new BasicNameValuePair(entry.getKey(), entry.getValue()));
+                    }
+                }
+
+                URIBuilder uriBuilder = new URIBuilder(String.format("%s%s", host.toURI(), path));
+                uriBuilder.addParameters(params);
+
+                httpPost = new HttpPost(uriBuilder.build());
+                httpPost.setConfig(transportRequestConfig);
+                httpPost.setHeader("X-SF-TOKEN", token);
+                httpPost.setHeader("User-Agent", USER_AGENT);
+                httpPost.setHeader("Content-Type", "text/plain");
+                if (body != null) {
+                    HttpEntity httpEntity = new StringEntity(body);
+                    httpPost.setEntity(httpEntity);
+                }
+
+                if (log.isDebugEnabled()) {
+                    log.debug(httpPost.toString());
+                }
+
+                CloseableHttpResponse response = client.execute(httpPost);
+
+                StatusLine statusLine = response.getStatusLine();
+                int statuscode = statusLine.getStatusCode();
+                if ((statuscode < 200) || (statuscode >= 300)) {
+
+                    try {
+                        response.close();
+                    } catch (IOException ex) {
+                        log.error("failed to close response", ex);
+                    }
+
+                    String errorMessage = statusLine.getStatusCode() + ": failed post [ " + httpPost
+                            + " ] reason: " + statusLine.getReasonPhrase();
+                    throw new SignalFlowException(statusLine.getStatusCode(), errorMessage);
+                }
+
+                return response;
+
+            } catch (IOException ex) {
+                throw new SignalFlowException(
+                        "failed post [ " + httpPost + " ] reason: " + ex.getMessage(), ex);
+            } catch (URISyntaxException ex) {
+                throw new SignalFlowException("failed post reason " + ex.getMessage(), ex);
```

Comment:
"failed post" seems like an obscure error message. The user doesn't necessarily know (or needs to know) that a HTTP POST is going on behind the scenes.